server/pulp/server/exceptions.py

```diff
@@ -92,7 +92,10 @@ def __init__(self, error_code=error_codes.PLP0001, **kwargs):
                                          field=key)
 
     def __str__(self):
-        msg = self.error_code.message % self.error_data
+        try:
+            msg = self.error_code.message % self.error_data
+        except ValueError:
+            msg = "Incorrectly formatted error string for {code}".format(code=self.error_code)
```

Comment:
This seems like it would obscure the real error message for the developer. I see that the intent is to try to help a developer out a bit, but I think this actually would be more difficult to understand. Compare:

```
>>> 'This is a %(test)' % {'test': 'test'}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: incomplete format
```

to:

```
>>> try:
...  'This is a %(test)' % {'test': 'test'}
... except ValueError:
...  print 'Incorrectly formatted error string for PLP-1234'
... 
Incorrectly formatted error string for PLP-1234
```

I think that ValueError: incomplete format is more informative, and it avoids dropping the stack trace which I think would be helpful during debugging. Also, I wonder if there's a way to trigger a ValueError by passing the wrong argument on the other side of the ```%``` operator. I tried to find something, but I was unsuccessful. If that is possible, this error message could be very misleading.

IMO, it's best to let developer facing errors happen without trying to intervene. For outside users, it is good to try to catch errors and give more human understandable meaning to them, but I think we can reasonably expect developers to be able to read ```ValueError: incomplete format``` and know what to do.